service/src/main/java/org/ehrbase/service/ContributionServiceImp.java

```diff
@@ -187,11 +190,45 @@ private void processCompositionVersion(UUID ehrId, UUID contributionId, Version
                 break;
             case SYNTHESIS:     // TODO
             case UNKNOWN:       // TODO
-            default:
+            default:    // TODO keep as long as above has TODOs. Check of valid change type is done in checkContributionRules
                 throw new UnexpectedSwitchCaseException(changeType);
         }
     }
 
+    /**
+     * Checks contribution rules, i.e. context-aware checks of the content. For instance, a committed version can't be
+     * of change type CREATION while containing a "preceding_version_uid".
+     *
+     * Note: Those rules are checked here, because context of the contribution might be important.
+     * Apart from that, most rules logically could be checked within the appropriate service as well.
+     * @param version Input version object
+     * @param changeType Change type of this version
+     */
+    private void checkContributionRules(Version version, I_ConceptAccess.ContributionChangeType changeType) {
+
+        switch (changeType) {
+            case CREATION:
+                // can't have change type CREATION and a given "preceding_version_uid"
+                if (version.getPrecedingVersionUid() != null)
+                    throw new InvalidApiParameterException("Invalid version. Change type CREATION, but also set \"preceding_version_uid\" attribute");
+                break;
+            case MODIFICATION:
+                // can't have change type MODIFICATION and without giving "preceding_version_uid"
+                if (version.getPrecedingVersionUid() == null)
+                    throw new InvalidApiParameterException("Invalid version. Change type MODIFICATION, but without \"preceding_version_uid\" attribute");
+                break;
+            // block of valid change types, without any rules to apply (yet)
+            case AMENDMENT:
```

Comment:
@jakesmolka amendment and modification have the same logic

Reply:
Thanks! Fixed in https://github.com/ehrbase/ehrbase/pull/202/commits/ace3e3abdd6ee8da505f49d6a6da87f994d55f45